fern/openapi-ir/definition/example.yml

```diff
@@ -0,0 +1,112 @@
+types:
+  SchemaInstanceId:
+    docs: |
+      Uniquely identifies an instance of a schema. For nested instances,
+      the id will be a dot-delimited path, e.g. foo.bar.baz.
+    type: string
+
+  Example:
+    docs: |
+      An OpenAPI example, which is either fully specified, partially
+      specified, or a reference to another schema instance's example.
+    union:
+      full: FullExample
+      partial: PartialExample
+      reference: ReferencedExample
+
+  FullOrPartialExample:
+    docs: |
+      Specifically designed for the PartialObjectExample.
+      The included properties can either be full or partial
+      examples, but not references.
+    union:
+      full: FullExample
+      partial: PartialExample
+
+  FullExample:
+    union:
+      primitive: PrimitiveExample
+      object: FullObjectExample
+      array: ArrayExample
+      map: MapExample
+      optional: FullExample
+      enum: EnumExample
+      reference: FullExample
+      literal: LiteralExample
+      oneOf: FullOneOfExample
+      nullable: FullExample
+      unknown: {}
+
+  PrimitiveExample:
+    union:
+      int: integer
+      int64: long
+      float: double
+      double: double
+      string: string
+      datetime: datetime
+      date: date
+      base64: base64
+      boolean: boolean
+
+  FullObjectExample:
+    properties:
+      properties: map<string, FullExample>
+
+  ArrayExample:
+    properties:
+      elements: list<FullExample>
+
+  MapExample:
+    properties:
+      pairs: list<MapExamplePair>
+
+  MapExamplePair:
+    properties:
+      key: PrimitiveExample
+      value: FullExample
+
+  EnumExample:
+    properties:
+      examples: list<EnumValueExample>
+
+  EnumValueExample:
+    properties:
+      value: string
+
+  FullOneOfExample:
+    union:
+      discriminated: FullDiscriminatedOneOfExample
+      undisciminated: FullUnDiscriminatedOneOfExample
+
+  FullDiscriminatedOneOfExample:
+    properties:
+      examples: map<string, FullExample>
```

Comment:
I think this should be 
```yaml
FullDiscriminatedOneOfExample: 
  properties: 
    properties: map<string, FullExample>
```

Reply:
would be down to inline as well 

```yaml
  PropertyKey: string
 
  FullOneOfExample:
    union:
      discriminated: map<PropertyKey, FullExample>
      undisciminated: FullExample
```

Reply:
Shouldn't this take the same form as the `PartialDiscriminatedOneOfExample`? Otherwise this ends up looking just like a `FullObjectExample`. I'm thinking something more along the lines of the following:

```yaml
  FullDiscriminatedOneOfExample:
    properties:
      discriminant: string
      value: FullExample
```

Reply:
Discussed offline - the full discriminated union will actually _always_ be an object, so we'll go with your suggestion here.